### PR TITLE
Python migration

### PR DESCRIPTION
**Pull request by AdamJHall**
This PR migrates this pipe to use python instead of a bash script (I've kept the bash script around for now) it is based on https://github.com/aligent/code-standards-pipe-php

I have removed the MAGENTO_USER and MAGENTO_PASS variables as realistically if you need advanced composer configuration you should do that in a separate step and pass that to this step as an artefact.

### CODE REVIEW

File: pipe/pipe.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+        if self.level:
+          phpstan_command.append(f"--level={self.level}")
+
+        phpstan = subprocess.run(
+          args=phpstan_command,
+          capture_output=True,
+          universal_newlines=True)
+
+        self.failure = False if phpstan.returncode == 0 else True
```
Comment (toddhainsworth) [inline pipe/pipe.py]: Thoughts?
```suggestion
        self.failure = phpstan.returncode != 0
```

**Review by toddhainsworth — APPROVED** [review]

File: pipe/pipe.py (hunk 1 above)
Comment (AdamJHall) [reply pipe/pipe.py]: Yeah that's a lot more readable

**Review by toddhainsworth — APPROVED** [review]

**Review by daniel-vdp — APPROVED** [review]
Looks good to me! Now that we're using pip we should probably add dependabot to the repo as well.

**Comment by AdamJHall** [comment]
Merged to staging, I'll setup some test PRs in bitbucket on Monday with the experimental tag to test again before pushing to main.

**Pull request merged**